src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c

```diff
@@ -21,30 +43,242 @@ const ActorInit Obj_Raillift_InitVars = {
+                    this->actionFunc = ObjRaillift_Teleport;
+                    func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+                    updatePos = false;
+                }
+            }
+        }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/ObjRaillift_Destroy.asm")
+        if (updatePos) {
+            ObjRaillift_UpdatePosition(this, this->curPoint);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19B98.asm")
+void ObjRaillift_Teleport(ObjRaillift* this, GlobalContext* globalCtx) {
+    // will teleport to what ever curpoint is set to
+    if (!func_800CAF70(&this->dyna)) {
+        ObjRaillift_UpdatePosition(this, this->curPoint);
+        func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->actionFunc = ObjRaillift_Move;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19BA8.asm")
+void ObjRaillift_Wait(ObjRaillift* this, GlobalContext* globalCtx) {
+    this->waitTimer--;
+    if (this->waitTimer <= 0) {
+        this->actionFunc = ObjRaillift_Move;
+        this->dyna.actor.speedXZ = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19E84.asm")
+void ObjRaillift_Idle(ObjRaillift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, OBJRAILLIFT_GET_FLAG(&this->dyna.actor))) {
+        this->dyna.actor.speedXZ = 0.0f;
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+        this->actionFunc = ObjRaillift_StartCutscene;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19EE0.asm")
+void ObjRaillift_StartCutscene(ObjRaillift* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->cutsceneTimer = 50;
+        this->actionFunc = ObjRaillift_Move;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19F18.asm")
+void ObjRaillift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRaillift* this = THIS;
+    f32 target;
+    f32 step;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19F78.asm")
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->dyna.actor, 10.0f);
+    if (this->cutsceneTimer > 0) {
+        this->cutsceneTimer--;
+        if (this->cutsceneTimer == 0) {
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+    if (OBJRAILLIFT_SHOULD_REACT_TO_WEIGHT(thisx)) {
+        this->hadWeightOn = this->hasWeightOn;
```

Comment:
```suggestion
        this->isWeightOnPrev = this->isWeightOn;
```
Maybe something like this to keep with boolean convention?